Tests/SentryTests/Profiling/SentryProfilerSwiftTests.swift

```diff
@@ -0,0 +1,373 @@
+        options.tracesSampleRate = 1.0
+
+        let expA = expectation(description: "Span A finishes")
+        let spanA = fixture.hub.startTransaction(name: fixture.transactionName, operation: fixture.transactionOperation)
+
+        // Some busy work to try and get it to show up in the profile.
+        let str = "a"
+        var concatStr = ""
+        for _ in 0..<100_000 {
+            concatStr = concatStr.appending(str)
+        }
+
+        DispatchQueue.main.asyncAfter(deadline: .now() + 45) {
+            spanA.finish()
+            expA.fulfill()
+        }
+
+        let expB = expectation(description: "Span B finishes")
+        DispatchQueue.main.asyncAfter(deadline: .now() + 35) {
+            let spanB = self.fixture.hub.startTransaction(name: self.fixture.transactionName, operation: self.fixture.transactionOperation)
+
+            // Some busy work to try and get it to show up in the profile.
+            let str = "a"
+            var concatStr = ""
+            for _ in 0..<100_000 {
+                concatStr = concatStr.appending(str)
+            }
+            
+            DispatchQueue.main.asyncAfter(deadline: .now() + 5) {
+                spanB.finish()
+                expB.fulfill()
+            }
+        }
+
+        waitForExpectations(timeout: 50)
+
+        XCTAssertEqual(self.fixture.client.captureEnvelopeInvocations.count, 2)
+        for envelope in self.fixture.client.captureEnvelopeInvocations.invocations {
+            XCTAssertEqual(1, envelope.items.count)
+            guard let profileItem = envelope.items.first else {
+                XCTFail("Expected at least 1 additional envelope item")
+                return
+            }
+            XCTAssertEqual("profile", profileItem.header.type)
+            self.assertValidProfileData(data: profileItem.data)
+        }
+    }
+
+    func testProfileTimeout() {
+        fixture.options.profilesSampleRate = 1.0
+        fixture.options.tracesSampleRate = 1.0
+        performTest(duration: 35)
+    }
+
+    func testStartTransaction_ProfilingDataIsValid() {
+        fixture.options.profilesSampleRate = 1.0
+        fixture.options.tracesSampleRate = 1.0
+        performTest()
+    }
+
+    func testProfilingDataContainsEnvironmentSetFromOptions() {
+        fixture.options.profilesSampleRate = 1.0
+        fixture.options.tracesSampleRate = 1.0
+        let expectedEnvironment = "test-environment"
+        fixture.options.environment = expectedEnvironment
+        performTest(transactionEnvironment: expectedEnvironment)
+    }
+
+    func testProfilingDataContainsEnvironmentSetFromConfigureScope() {
+        fixture.options.profilesSampleRate = 1.0
+        fixture.options.tracesSampleRate = 1.0
+        let expectedEnvironment = "test-environment"
+        fixture.hub.configureScope { scope in
+            scope.setEnvironment(expectedEnvironment)
+        }
+        performTest(transactionEnvironment: expectedEnvironment)
+    }
```

Comment:
These were refactored to extract most of their bodies into the new function `performTest`